mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala

```diff
@@ -99,19 +106,27 @@ object Keyword {
     Keyword("throw", isExpression = true),
     Keyword("implicit", isBlock = true, isTemplate = true),
     Keyword("return", isMethodBody = true),
+    Keyword("extends", leadingReverseTokens = Some(leadingTokensExtend)),
     Keyword("match"), // already implemented by CompletionPosition
     Keyword("case"), // already implemented by CompletionPosition and "case class"
     Keyword("override"), // already implemented by CompletionPosition
     Keyword("forSome"), // in-frequently used language feature
     Keyword("macro"), // in-frequently used language feature
     // The keywords below were left out in the first iteration of implementing keyword completions
     // since they appear in positions that are a bit more difficult to detect on the syntax tree.
-    Keyword("extends"),
     Keyword("with"),
     Keyword("catch"),
-    Keyword("extends"),
     Keyword("finally"),
     Keyword("then")
   )
 
+  private def leadingTokensExtend(leadingReverseTokens: Iterator[Token]): Boolean = {
+    leadingReverseTokens.filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF]).take(3).toList match {
+      // (class|trait|object) classname ext@@
+      case (_: Token.Ident) :: (_: Token.Ident) :: kw :: Nil =>
```

Comment:
It's because I'm doing `take(3)`, so there shouldn't be elements anymore. `case (_: Token.Ident) :: (_: Token.Ident) :: kw :: _ =>` might be easier to understand?

Reply:
Right, `take(3)`, now it makes perfect sense, there's no need to change